src/GitHubCreateMergePRs/Program.cs

```diff
@@ -157,4 +156,53 @@ private static async Task RunAsync(XDocument config, bool isAutomatedRun, bool i
             }
         }
     }
+
+    private static bool ShouldRunMerge(XElement merge, bool isAutomatedRun)
+    {
+        // We always run when merges are started manually
+        if (!isAutomatedRun)
+        {
+            return true;
+        }
+
+        var frequency = merge.Attribute("frequency")?.Value.ToLower();
+
+        // We always run when a frequency isn't specified
+        if (string.IsNullOrEmpty(frequency))
+        {
+            return true;
+        }
+
+        // We always run when an unexpected frequency value is specified
+        if (frequency != "daily" && frequency != "weekly")
+        {
+            Console.WriteLine($"##vso[task.logissue type=error]Unexpected merge frequency specified ({frequency}).");
+
+            return true;
+        }
+
+        var runDateTime = DateTime.Now;
+        var currentDate = runDateTime.Date;
+
+        // Are we within the first 15 minutes of the day?
+        var fifteenMinutes = new TimeSpan(hours: 0, minutes: 15, seconds: 0);
+        var isStartOfDay = runDateTime - currentDate < fifteenMinutes;
```

Comment:
If the machine time is running fast, should we allow 15min leeway into the previous day as well?

Reply:
That seems excessive. =p

Reply:
Although I could take the `runDateTime` and `.AddMinutes(5)` then do all my calculations based on the adjusted DateTime to keep things simple...

Reply:
But really I just hope that triggers don't fire before the trigger time...